youtube_dl/extractor/picosong.py

```diff
@@ -0,0 +1,47 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import urljoin
+
+
+class PicosongIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?picosong\.com/(?P<id>[0-9A-Za-z]{4})'
+    _TEST = {
+        'url': 'http://picosong.com/7uer/',
+        'md5': '7f2ee011ba3ca2ac180a35deb10feabf',
+        'info_dict': {
+            'id': '7uer',
+            'ext': 'mp3',
+            'artist': 'Xarlable',
+            'title': 'Rock My Forum',
+            'album': 'Epic',
+            'genre': 'Howto & Style',
+        }
+    }
+
+    def _extract_info_td(self, page, key):
+        # <td class="key">Title:</td>
+        # <td class="value">Rock My Forum</td>
+        return self._html_search_regex(
+            r'%s:</td>\s*<td class="value">([^<]+)' % (key.capitalize(), ),
+            page, key)
```

Comment:
This will break extraction if some field is missing. Optional fields should not break extraction.